blackd.py

```diff
@@ -0,0 +1,116 @@
+
+import black
+import click
+
+
+@click.command(context_settings={"help_option_names": ["-h", "--help"]})
+@click.option(
+    "-l",
+    "--line-length",
+    type=int,
+    default=black.DEFAULT_LINE_LENGTH,
+    help="How many character per line to allow.",
+    show_default=True,
+)
+@click.option(
+    "--py36",
+    is_flag=True,
+    help=(
+        "Allow using Python 3.6-only syntax on all input files.  This will put "
+        "trailing commas in function signatures and calls also after *args and "
+        "**kwargs.  [default: per-file auto-detection]"
+    ),
+)
+@click.option(
+    "--pyi",
+    is_flag=True,
+    help=(
+        "Format all input files like typing stubs regardless of file extension "
+        "(useful when piping source on standard input)."
+    ),
+)
+@click.option(
+    "-S",
+    "--skip-string-normalization",
+    is_flag=True,
+    help="Don't normalize string quotes or prefixes.",
+)
+@click.option(
+    "--fast/--safe",
+    is_flag=True,
+    help="If --fast given, skip temporary sanity checks. [default: --safe]",
+)
+@click.option(
+    "--config",
+    type=click.Path(
+        exists=False, file_okay=True, dir_okay=False, readable=True, allow_dash=False
+    ),
+    is_eager=True,
+    callback=black.read_pyproject_toml,
+    help="Read configuration from PATH.",
+)
+@click.version_option(version=black.__version__)
+@click.pass_context
+def main(
+    ctx: click.Context,
+    line_length: int,
+    fast: bool,
+    pyi: bool,
+    py36: bool,
+    skip_string_normalization: bool,
+    config: Optional[str],
+) -> None:
+    mode = black.FileMode.from_configuration(
+        py36=py36, pyi=pyi, skip_string_normalization=skip_string_normalization
+    )
+    run_server(line_length=line_length, fast=fast, mode=mode)
+    black.out("done")
+
+
+def run_server(line_length: int, fast: bool, mode: black.FileMode) -> None:
+    loop = asyncio.get_event_loop()
+    req_handler = partial(new_req, line_length=line_length, mode=mode, fast=fast)
+    server = loop.run_until_complete(
+        asyncio.start_server(req_handler, host="localhost", port=8888)
+    )
+    try:
+        loop.run_forever()
+    finally:
+        server.close()
+        loop.run_until_complete(server.wait_closed())
+
+
+async def new_req(
+    reader: asyncio.StreamReader,
+    writer: asyncio.StreamWriter,
+    line_length: int,
+    fast: bool,
+    mode: black.FileMode,
+) -> None:
+    try:
+        peer = writer.get_extra_info("peername")
+        req_bytes = await reader.read()
+        black.out(f"[{peer}] got {len(req_bytes)} bytes")
+        req_str = req_bytes.decode("utf8")
+        formatted = black.format_file_contents(
+            req_str, line_length=line_length, fast=fast, mode=mode
+        ).encode("utf8")
+        writer.write(b"OK\n" + formatted)
+        black.out(f"[{peer}] format done, sending {len(formatted)} bytes")
+    except black.NothingChanged:
+        writer.write(b"OK\n" + req_bytes)
```

Comment:
Would it be cleaner to use a different status string for this case? There could be a file that just only contains an int literal.